src/cli/common.rs

```diff
@@ -291,24 +291,20 @@ pub(crate) async fn update_all_channels(
         info!("no updatable toolchains installed");
     }
 
-    let show_channel_updates = || {
-        if !toolchains.is_empty() {
-            writeln!(cfg.process.stdout().lock())?;
-
-            let t = toolchains
-                .into_iter()
-                .map(|(p, s)| (PackageUpdate::Toolchain(p), s))
-                .collect();
-            show_channel_updates(cfg, t)?;
-        }
-        Ok(())
-    };
+    if !toolchains.is_empty() {
+        writeln!(cfg.process.stdout().lock())?;
+
+        let t = toolchains
+            .into_iter()
+            .map(|(p, s)| (PackageUpdate::Toolchain(p), s))
+            .collect();
+        show_channel_updates(cfg, t)?;
```

Comment:
Sorry, but I'm a bit confused about this... Do you know why are we showing channel updates during `self update` in the first place?

Reply:
@djc If you are too occupied, I can offer to do this for you on your permission :)